rflx/specification/parser.py

```diff
@@ -903,15 +903,28 @@ def create_modular(
     _workers: int,
     _cache: Cache,
     filename: Path,
-) -> Optional[model.Type]:
+) -> None:
     assert isinstance(modular, lang.ModularTypeDef)
-    result = model.ModularInteger(
-        identifier,
-        create_math_expression(error, modular.f_mod, filename),
-        type_location(identifier, modular),
+    modulus = create_math_expression(error, modular.f_mod, filename).simplified()
+    assert isinstance(modulus, expr.Number)
+    upper = int(modulus) - 1
+    error.extend(
+        [
+            (
+                "modular integer types are not supported",
+                Subsystem.PARSER,
+                Severity.ERROR,
+                type_location(identifier, modular),
+            ),
+            (
+                f'use "type {identifier.name} is range 0 .. {upper}'
+                f' with Size => {upper.bit_length()}" instead',
+                Subsystem.PARSER,
+                Severity.INFO,
+                type_location(identifier, modular),
+            ),
+        ]
```

Comment:
Great idea to keep the modular integer and even give the user something to copy-and-past!